nowledge-mem-codex-plugin/hooks/nmem-stop-save.py

```
@@ -0,0 +1,324 @@
+            state = load_json(STATE_FILE)
+            state[session_id] = {
+                "content_hash": content_hash,
+                "message_count": message_count,
+                "thread_id": thread_id,
+                "source_file": str(transcript_path),
+                "updated_at": datetime.now(timezone.utc).isoformat(),
+            }
+            save_json(STATE_FILE, state)
+    except TimeoutError as exc:
+        return 0, f"skip: failed to lock state file for session={session_id}: {exc}"
+    except Exception as exc:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)